lib/logstash/filters/translate.rb

```diff
@@ -137,18 +137,18 @@ def load_yaml(registering=false)
 
     begin
       @dictionary.merge!(YAML.load_file(@dictionary_path))
-    rescue Exception => e
```

Comment:
I know SyntaxError is a RuntimeError, because in ruby is a ruby syntax error so it happen in runtime, not an error type located for things like YAML, JSON, or XML syntax validation.

The question I ask myself here is, the actual behaviour is if registering is set to true, raise the exception otherwise not. If we aim to keep this, adding ```YAML::SyntaxError``` provide us only with a way to detect when this is a syntax error or not. I understand we don't want to react different for SyntaxError and other errors (StandardError). In your example you aim to raise only SyntaxException up, and log everything under StandardError.

I understand the issue as not reporting __everything__ as syntax error even when they are not, in my example user will notice when Psych claims an error, and also when there are other error. 

What do you think?